### DISCUSSION TITLE
Deleting durable consumers when calling Drain()

### DISCUSSION THREAD

**Discussion by britishdan** [nats-io/nats.go#1110] [General]
Hey!
We were (very) surprised to find out that when calling `Drain()`, the nats client **deletes durable consumers** that it created during `QueueSubscribe()`.
It took us a good couple of days to figure out why our durable consumers were "disappearing" 🙈 

I was wondering what the reason is for this behavior?
It seems counter intuitive to us.
We assumed that `Drain()` would stop consuming messages, but not delete consumers, especially **durable** consumers.

Thank you.

**Comment by derekcollison** [comment]
We have been reviewing these things internally and have been working on some updates.

For background, if the app creates the durable through a Subscribe call the code makes the assumption that if you unsubscribe or drain you want it to go away.

You can fix this by calling AddConsumer and then binding the subscribe to the durable consumer in your code.

Thanks for the feedback!

/cc @piotrpio @wallyqs

**Reply by britishdan** [reply]
Thanks for the reply! The call to AddConsumer worked.

I think it's great that you're rethinking this assumption 😄 
Our assumption was that durable is... well... durable.
